### ISSUE TITLE
Inconsistent handling of terraform.tfvars issues when --recursive is used

### ISSUE THREAD

**Issue by tbutler-qontigo** [terraform-linters/tflint#2368]
### Summary

If I have an "invalid" `terraform.tfvars` file then the output is different depending on the use of `--recursive` or not

Here is my `terraform.tfvars` file:
```
foo = "bar"
baz = #{buzz}
```
`tflint --format=sarif` gives me this:
```json
     "results": [
        {
          "ruleId": "Missing expression",
          "ruleIndex": 18446744073709551615,
          "level": "error",
          "message": {
            "text": "Expected the start of an expression, but found the end of the file."
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "terraform.tfvars"
                },
                "region": {
                  "startLine": 2,
                  "startColumn": 14,
                  "endLine": 2,
                  "endColumn": 14,
                  "byteOffset": 26,
                  "byteLength": 0
                }
              }
            }
          ]
        }
      ]
```
Which looks reasonable, (apart from the ruleIndex which I have raised in #2367)
If I add `--recursive` then `tflint --recursive --format=sarif` gives me this:
```json
      "results": [
        {
          "ruleId": "application_error",
          "ruleIndex": 18446744073709551615,
          "level": "error",
          "message": {
            "text": "Failed to run in .; exit status 1\n\nFailed to load values files; terraform.tfvars:2,14-14: Missing expression; Expected the start of an expression, but found the end of the file.:\n\n\u001b[31mError\u001b[0m: Missing expression\n\n  on terraform.tfvars line 2:\n   2: baz = #{buzz}\n\nExpected the start of an expression, but found the end of the file.\n\n"
          }
        }
      ]
```
Which is rather less useful.  Clearly it does know where the problem is since the error text includes all the details, but I would expect the same sort of output as when I do not specify `--recursive`.

Separately I would like to be able to ignore this invalid expression message in `terraform.tfvars` specifically (we substitute in real values prior to actually using the terraform) - is there a way to do this

Thanks

### Command

tflint --recursive --format=sarif

### Terraform Configuration

```terraform
terraform {
  backend "azurerm" {
  }
}
```

### TFLint Configuration

```HCL
plugin "terraform" {
  enabled = true
  preset  = "all"
}

plugin "azurerm" {
  enabled = true
  version = "0.28.0"
  source  = "github.com/terraform-linters/tflint-ruleset-azurerm"
}
```

### Output

```Shell
{
  "version": "2.1.0",
  "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/main/sarif-2.1/schema/sarif-schema-2.1.0.json",
  "runs": [
    {
      "tool": {
        "driver": {
          "informationUri": "https://github.com/terraform-linters/tflint",
          "name": "tflint",
          "rules": [],
          "version": "0.58.1"
        }
      },
      "results": []
    },
    {
      "tool": {
        "driver": {
          "informationUri": "https://github.com/terraform-linters/tflint",
          "name": "tflint-errors",
          "rules": [],
          "version": "0.58.1"
        }
      },
      "results": [
        {
          "ruleId": "application_error",
          "ruleIndex": 18446744073709551615,
          "level": "error",
          "message": {
            "text": "Failed to run in .; exit status 1\n\nFailed to load values files; terraform.tfvars:2,14-14: Missing expression; Expected the start of an expression, but found the end of the file.:\n\n\u001b[31mError\u001b[0m: Missing expression\n\n  on terraform.tfvars line 2:\n   2: baz = #{buzz}\n\nExpected the start of an expression, but found the end of the file.\n\n"
          }
        }
      ]
    }
  ]
}
```

### TFLint Version

0.58.1

### Operating System

- [ ] Linux
- [ ] macOS
- [x] Windows

**Comment by wata727** [comment]
This issue will be fixed as part of #1669.

> Separately I would like to be able to ignore this invalid expression message in terraform.tfvars specifically (we substitute in real values prior to actually using the terraform) - is there a way to do this

There is no such solution. We recommend that you run a preprocessing before running TFLint, similar to what you do with Terraform.
